### PR TITLE
adds no_proxy

### PR DESCRIPTION
**Pull request by javierwilson**
whoever honors http_proxy *should* also honor no_proxy.
adding no_proxy for all hosts in inventory.

### CODE REVIEW

**Review by emord — APPROVED** [review]
thanks @javierwilson

File: fab/operations/supervisor.py [hunk 1]
```diff
@@ -253,9 +253,12 @@ def _format_env(current_env, extra=None):
         ret['supervisor_env_vars']['NEW_RELIC_CONFIG_FILE'] = '%(root)s/newrelic.ini' % env
         ret['supervisor_env_vars']['NEW_RELIC_ENVIRONMENT'] = '%(environment)s' % env
 
+    all_hosts = [host.name for host in InventoryParser(env.inventory).groups['all'].get_hosts()]
+
     if env.http_proxy:
         ret['supervisor_env_vars']['http_proxy'] = 'http://{}'.format(env.http_proxy)
         ret['supervisor_env_vars']['https_proxy'] = 'https://{}'.format(env.http_proxy)
+        ret['supervisor_env_vars']['no_proxy'] = '{},relay.nic.in,smsgw.sms.gov.in,icds-cas.gov.in'.format(','.join(all_hosts))
```
Comment (benrudolph) [inline fab/operations/supervisor.py]: do you think we can put these extra hosts in a variable in `environments.yml`? something like `additional_no_proxy_hosts` maybe?

**Review by benrudolph — COMMENTED** [review]
nice, one small comment. thanks for the quick turnaround

File: fab/operations/supervisor.py (hunk 1 above)
Comment (javierwilson) [reply fab/operations/supervisor.py]: so taht they only affect the specific environment? that would make sense

**Comment by javierwilson** [comment]
it now uses additional_no_proxy_hosts from environment.yml

**Review by emord — APPROVED** [review]
Looks good to me @javierwilson is there any reason to not merge this? it still has the don't merge label on

**Review by benrudolph — APPROVED** [review]

**Pull request merged**